api/leaderworkerset/v1/leaderworkerset_types.go

```diff
@@ -93,6 +93,11 @@ type LeaderWorkerSetSpec struct {
 	// RolloutStrategy defines the strategy that will be applied to update replicas
 	// when a revision is made to the leaderWorkerTemplate.
 	RolloutStrategy RolloutStrategy `json:"rolloutStrategy"`
+
+	// StartupPolicy determines the startup order of the leader statefulSet and worker statefulSets.
+	// +kubebuilder:default=LeaderCreated
+	// +kubebuilder:validation:Enum={LeaderCreated,LeaderReady}
+	StartupPolicy StartupPolicyType `json:"startupPolicy"`
```

Comment:
As we described in https://github.com/kubernetes-sigs/lws/issues/147, let's make it an optional field.